### PR TITLE
Fix Messages' bodies that are re-downloaded each time a Thread is opened

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #137

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/api/MailApi.kt [hunk 1]
```diff
@@ -50,10 +50,10 @@ object MailApi {
         return ApiRepository.getThreads(mailboxUuid, folder.id).data?.threads?.map { it.initLocalValues() }
     }
 
-    fun fetchMessages(thread: Thread): List<Message> {
+    fun fetchMessages(thread: Thread): List<Pair<Message, Boolean>> {
         return thread.messages.map { realmMessage ->
             if (realmMessage.fullyDownloaded) {
-                realmMessage
+                realmMessage to true
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/api/MailApi.kt:56]: This realm object is not a managed realm ?

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/api/MailApi.kt:56]: Oh yes it is! I don't need this boolean, thanks.

File: app/src/main/java/com/infomaniak/mail/data/MailData.kt [hunk 2]
```diff
@@ -366,22 +374,29 @@ object MailData {
...
+                            }
+                    }
+                    copyToRealm(apiThread, UpdatePolicy.ALL)
+                }
+            }
+
+            val liveFolder = getLatestFolder(folder.id) ?: folder
+            liveFolder.threads = apiThreads.toRealmList()
+            copyToRealm(liveFolder, UpdatePolicy.ALL)
+        }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/MailData.kt:399]: Need refactor

File: app/src/main/java/com/infomaniak/mail/data/MailData.kt [hunk 3]
```diff
@@ -366,22 +374,29 @@ object MailData {
...
-        //                     attachments = realmMessage.attachments
-        //                 }
-        //             }
-        //         }
-        //     } ?: apiThread
-        // }.toRealmList()
-        MailboxContentController.upsertFolder(folder)
+        MailRealm.mailboxContent.writeBlocking {
+            apiThreads.forEach { apiThread ->
+                realmThreads?.find { it.uid == apiThread.uid }?.let { realmThread ->
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/MailData.kt:379]: use realm query

File: app/src/main/java/com/infomaniak/mail/data/MailData.kt [hunk 4]
```diff
@@ -366,22 +374,29 @@ object MailData {
...
+                    apiThread.messages.forEach { apiMessage ->
+                        realmThread.messages.find { it.uid == apiMessage.uid }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/MailData.kt:381]: use realm query

File: app/src/main/java/com/infomaniak/mail/data/MailData.kt [hunk 5]
```diff
@@ -366,22 +374,29 @@ object MailData {
...
+                            ?.let { getLatestMessage(it.uid) }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/MailData.kt:382]: You can simply use `getLatestMessage` without make this `realmThread.messages.find { it.uid == apiMessage.uid }`

File: app/src/main/java/com/infomaniak/mail/data/MailData.kt [hunk 6]
```diff
@@ -366,22 +374,29 @@ object MailData {
...
+                                    attachments = realmMessage.attachments
+                                }
+                                copyToRealm(apiMessage, UpdatePolicy.ALL)
+                            }
+                    }
+                    copyToRealm(apiThread, UpdatePolicy.ALL)
+                }
+            }
+
+            val liveFolder = getLatestFolder(folder.id) ?: folder
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/MailData.kt:396]: rename `liveFolder` to `latestFolder`

**Review by sirambd — CHANGES_REQUESTED** [review]
Waiting for request changes

**Pull request merged**